sycl/doc/extensions/proposed/sycl_ext_intel_fpga_pipes_properties.asciidoc

```diff
@@ -0,0 +1,280 @@
+implementation can include more capacity than this parameter, but not less.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+
+|`ready_latency`
+| Valid values: Non-negative integer value.
+
+Default value: 0
+
+The number of cycles between when the ready signal is deasserted and when the
+pipe can no longer accept new inputs.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`bits_per_symbol`
+| Valid values: A positive integer value that evenly divides by the data type size. 
+
+Default value: Datatype size
+
+Describes how the data is broken into symbols on the data bus.
+
+Data is broken down according to how you set the first_symbol_in_high_order_bits
+property. By default, data is broken down in little endian order.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`uses_valid`
+| Valid values: true or false
+
+Default value: true
+
+Controls whether a valid signal is present on the pipe interface. If false, the
+upstream source must provide valid data on every cycle that ready is asserted.
+
+This is equivalent to changing the pipe read calls to tryRead and assuming that
+success is always true.
+
+If set to false, min_capacity and ready_latency must be 0.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`uses_ready`
+| Valid values: true or false
+
+Default value: true
+
+Controls whether a ready signal is present. If false, the downstream sink must
+be able to accept data on every cycle that valid is asserted. This is
+equivalent to changing the pipe read calls to tryWrite and assuming that success
+is always true.
+
+If set to false, ready_latency must be 0.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`in_csr`
+| Valid Values: true or false
+
+Default Value: false
+
+Controls whether the host pipe is implemented using the Control and Status register (CSR).
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`first_symbol_in_high_order_bits`
+| Valid values: true or false
+
+Default value: false
+
+Specifies whether the data symbols in the pipe are in big-endian
+order.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`protocol`
+| Specifies the protocol for the pipe interface.
+Currently, the only protocol supported is `avalon`.
```

Comment:
It seems strange to me to have a property with only one legal value that is also the default.  I understand we're trying to leave the door open for other protocols in the future, but if that happens we could add the property at that time.

It also seems weird to present protocol as its own independent property that is orthogonal to the other properties when a bunch of the other properties (bits_per_symbol, first_symbol_in_high_order_bits and ready_latency) are properties of the avalon ST spec and don't necessarily apply to other interface protocols.  For example, AXI doesn't seem to have an endianness property so first_symbol_in_high_order_bits doesn't make sense with that protocol.  Not to mention "first_symbol_in_high_order_bits" is the name pulled directly from the avalon spec and if we were trying to be generic we would call it "endianness" instead.  

I see four possible solutions:

1. Drop the protocol property completely and accept that for now all our properties are avalon-centric.  This is the least work now but might require the most refactoring in the future to support other protocols.  
2. Make the protocol-dependent properties _part of the protocol property_.  This way different protocols can have different sub-properties as determined by their various protocol specifications.  
3. Try to come up with generic names for every property that will apply to any possible future protocol.  I think this is doomed to failure (see the endianness AXI interfaction).
4. Keep separate properties like we have in this PR but add extra restrictions to the protocol-dependent properties that say: "this property is only valid if specified in conjunction with the protocol<avalon> property"